lib/public_key/src/public_key.erl

```diff
@@ -1228,6 +1232,41 @@ pkix_test_data(#{} = Chain) ->
 pkix_test_root_cert(Name, Opts) ->
     pubkey_cert:root_cert(Name, Opts).
 
+%%--------------------------------------------------------------------
+-spec ocsp_status(binary(), list(), undefined | binary()) ->
+    {ok, term()} | {error, term()}.
+%%
+%% Description: Verify the OCSP response to get the certificate status
+%%--------------------------------------------------------------------
+ocsp_status(OCSPResponseDer, ResponderCerts, Nonce) ->
+    catch pubkey_ocsp:verify_ocsp_response(
+        OCSPResponseDer, ResponderCerts, Nonce).
+
+%%--------------------------------------------------------------------
+-spec ocsp_extensions(undefined | binary()) -> list().
+%% Description: Get OCSP stapling extensions for request
+%%--------------------------------------------------------------------
+ocsp_extensions(Nonce) ->
+    [Extn || Extn <- [pubkey_ocsp:get_nonce_extn(Nonce),
+                      pubkey_ocsp:get_acceptable_response_types_extn()],
+             erlang:is_record(Extn, 'Extension')].
+
+%%--------------------------------------------------------------------
+-spec get_ocsp_responder_id(#'Certificate'{}) -> binary().
+%%
+%% Description: Get the OCSP responder ID der
+%%--------------------------------------------------------------------
+get_ocsp_responder_id(Cert) ->
+    pubkey_ocsp:get_ocsp_responder_id(Cert).
+
+%%--------------------------------------------------------------------
+-spec get_nonce(boolean()) -> binary() | undefined.
+%%
+%% Description: Get an OCSP nonce
+%%--------------------------------------------------------------------
+get_nonce(OcspNonce) ->
```

Comment:
I do not think we need to have such an API function in public_key. I think the generating of the nonce should be handled in the ssl code and inputted into  API functions above.

Reply:
I put this comment here so that we won't forget to DER encode the nonce according to RFC6066.

> In the case of the "id-pkix-ocsp-nonce" OCSP extension, [RFC2560] is
unclear about its encoding; for clarification, the nonce MUST be a
DER-encoded OCTET STRING, which is encapsulated as another OCTET
STRING (note that implementations based on an existing OCSP client
will need to be checked for conformance to this requirement).

Reply:
For the DER encoding, we want to use a public_key function. I think we should be able to use
public_key:der_encode/2. I am not sure if we need to create a special  ASN1 type spec to solve it or if we can use an existing one.

Reply:
> I do not think we need to have such an API function in public_key. I think the generating of the nonce should be handled in the ssl code and inputted into API functions above.

maybe change the name to ocsp_nonce in public_key? since we may need encode the nonce first in public_key according to Peter's comment.

Reply:
> I put this comment here so that we won't forget to DER encode the nonce according to RFC6066.
> 
> > In the case of the "id-pkix-ocsp-nonce" OCSP extension, [RFC2560] is
> > unclear about its encoding; for clarification, the nonce MUST be a
> > DER-encoded OCTET STRING, which is encapsulated as another OCTET
> > STRING (note that implementations based on an existing OCSP client
> > will need to be checked for conformance to this requirement).

done